model/Scout.java

```diff
@@ -1,5 +1,153 @@
 package model;
 
-public class Scout {
-    
+import java.sql.SQLException;
+import java.util.Date;
+import java.util.Enumeration;
+import java.util.Properties;
+import java.util.Vector;
+
+import exception.InvalidPrimaryKeyException;
+
+public class Scout extends EntityBase
+{
+	private static final String myTableName = "Scout";
+
+	protected Properties dependencies;
+
+	// GUI Components
+
+	private String updateStatusMessage = "";
+	
+	public Scout(String scoutId) throws InvalidPrimaryKeyException
+	{
+		super(myTableName);
+
+		setDependencies();
+		persistentState = new Properties();
+		lookupAndStore(scoutId);
+	}
+	
+	public Scout()
+	{
+		super(myTableName);
+		
+		setDependencies();
+		persistentState = new Properties();
+		persistentState.setProperty("LastName", "");
+		persistentState.setProperty("FirstName", "");
+		persistentState.setProperty("MiddleName", "");
+		persistentState.setProperty("DateOfBirth", "");
+		persistentState.setProperty("PhoneNumber", "");
+		persistentState.setProperty("Email", "");
+		persistentState.setProperty("TroopID", "");
+	}
+	
+	public Scout(Properties props)
+	{
+		super(myTableName);
+		
+		setDependencies();
+		persistentState = new Properties();
+		Enumeration allKeys = props.propertyNames();
+		while (allKeys.hasMoreElements())
+		{
+			String nextKey = (String)allKeys.nextElement();
+			String nextValue = props.getProperty(nextKey);
+
+			if (nextValue != null) persistentState.setProperty(nextKey, nextValue);
+		}
+	}
+	
+	public void lookupAndStore(String condition) throws InvalidPrimaryKeyException
+	{
+		Vector<Properties> allDataRetrieved = getSelectQueryResult("SELECT * FROM " + myTableName + " WHERE (" + condition + ")");
+
+		// You must get one scout at least
+		if (allDataRetrieved == null) throw new InvalidPrimaryKeyException("No scout matching condition \"" + condition + "\" found.");
+		
+		// There should be EXACTLY one scout. More than that is an error
+		if (allDataRetrieved.size() != 1) throw new InvalidPrimaryKeyException("Multiple scouts matching condition \"" + condition + "\" found.");
+		
+		// copy all the retrieved data into persistent state
+		Properties retrievedScoutData = allDataRetrieved.elementAt(0);
+		persistentState.clear();
+
+		Enumeration allKeys = retrievedScoutData.propertyNames();
+		while (allKeys.hasMoreElements())
+		{
+			String nextKey = (String)allKeys.nextElement();
+			String nextValue = retrievedScoutData.getProperty(nextKey);
+
+			if (nextValue != null) persistentState.setProperty(nextKey, nextValue);
+		}
+	}
+	
+	public void update() 
+	{
+		try
+		{
+			String id = persistentState.getProperty("ID"); 
+			if (id != null)
+			{
+				Properties whereClause = new Properties();
+				whereClause.setProperty("ID", id);
+				updatePersistentState(mySchema, persistentState, whereClause);
+				updateStatusMessage = "Scout data for scout ID : " + id + " updated successfully in database!";
```

Comment:
Primary key references (id's) shouldn't be shown in the GUI. Instead display their name